components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementServiceTest.java

```diff
@@ -0,0 +1,289 @@
+                    " already registered");
+        }
+    }
+
+    @Test
+    public void registerOAuthApplicationWithNewSPNoRedirectUri() throws NoSuchFieldException,
+            IllegalAccessException, DCRException, IdentityApplicationManagementException {
+        registerOAuthApplication1();
+        mockApplicationManagementService = mock(ApplicationManagementService.class);
+
+        DCRDataHolder dcrDataHolder = DCRDataHolder.getInstance();
+        dcrDataHolder.setApplicationManagementService(mockApplicationManagementService);
+
+         when(mockApplicationManagementService.getServiceProvider(applicationName, tenantDomain)).thenReturn(null,
+                 new ServiceProvider());
+        try {
+            dcrManagementService.registerOAuthApplication(mockRegistrationRequestProfile);
+        } catch (IdentityException ex) {
+            assertEquals(ex.getMessage(), "RedirectUris property must have at least one URI value.");
+        }
+    }
+
+    @Test
+    public void registerOAuthApplicationWithNewSPWithFragmentRedirectUri() throws NoSuchFieldException,
+            IllegalAccessException, DCRException, IdentityApplicationManagementException, IdentityValidationException {
+        registerOAuthApplication1();
+
+        redirectUris.add("wvuv#");
+
+        String redirectUri = redirectUris.get(0);
+        mockApplicationManagementService = mock(ApplicationManagementService.class);
+
+        DCRDataHolder dcrDataHolder = DCRDataHolder.getInstance();
+        dcrDataHolder.setApplicationManagementService(mockApplicationManagementService);
+
+        when(mockApplicationManagementService.getServiceProvider(applicationName, tenantDomain)).thenReturn(null,
+                new ServiceProvider());
+
+        when(mockRegistrationRequestProfile.getRedirectUris()).thenReturn(redirectUris);
+
+        try {
+            dcrManagementService.registerOAuthApplication(mockRegistrationRequestProfile);
+        } catch (IdentityException ex) {
+            assertEquals(ex.getMessage(), "Redirect URI: " + redirectUri + ", is invalid");
+        }
+    }
+
+    @DataProvider(name = "size")
+    public Object[][] getSize() {
+        return new Object[][]{
+                {"0", ""},
+                {"1", dummyOauthConsumerSecret},
+                {"2", null}
+        };
+    }
+
+    @Test(dataProvider = "size")
+    public void registerOAuthApplicationWithNewSPWithRedirectUri(String listSize, String oauthConsumerSecret)
+            throws Exception {
+        registerOAuthApplication1();
+
+        if (Integer.valueOf(listSize) == 1) {
+            redirectUris.clear();
+            redirectUris.add("redirectUri1");
+        } else if (Integer.valueOf(listSize) == 2){
+            redirectUris.clear();
+            redirectUris.add("redirectUri1");
+            redirectUris.add("redirectUri2");
+        } else {
+            redirectUris.clear();
+        }
```

Comment:
Have this logic in the data provider and pass the redirectUris directly as the parameter